scripts/benchmark/utils/chain_plumbing.py

```diff
@@ -0,0 +1,131 @@
+from typing import (
+    Any,
+    Callable,
+    Iterable,
+    NamedTuple,
+    Type
+)
+
+from eth_keys import (
+    keys
+)
+
+from eth_utils import (
+    decode_hex,
+    to_wei,
+)
+
+from eth_typing import (
+    Address
+)
+
+from evm import (
+    constants,
+    Chain
+)
+from evm.vm.base import (
+    BaseVM
+)
+from evm.chains.mainnet import (
+    MainnetChain
+)
+from evm.db.backends.memory import (
+    MemoryDB
+)
+
+AddressSetup = NamedTuple('AddressSetup', [
+    ('address', Address),
+    ('balance', int),
+    ('code', bytes)
+])
+
+ALL_VM = [vm for _, vm in MainnetChain.vm_configuration]
+
+FUNDED_ADDRESS_PRIVATE_KEY = keys.PrivateKey(
+    decode_hex('0x45a915e4d060149eb4365960e6a7a45f334393093061116b197e3240065ff2d8')
+)
+
+FUNDED_ADDRESS = Address(FUNDED_ADDRESS_PRIVATE_KEY.public_key.to_canonical_address())
+
+DEFAULT_INITIAL_BALANCE = to_wei(1000, 'ether')
+
+SECOND_EXISTING_ADDRESS = Address(b'\0' * 19 + b'\x02')
+
+NON_EXISTING_ADDRESS = Address(b'\0' * 19 + b'\x03')
+
+GENESIS_PARAMS = {
+    'parent_hash': constants.GENESIS_PARENT_HASH,
+    'uncles_hash': constants.EMPTY_UNCLE_HASH,
+    'coinbase': constants.ZERO_ADDRESS,
+    'transaction_root': constants.BLANK_ROOT_HASH,
+    'receipt_root': constants.BLANK_ROOT_HASH,
+    'difficulty': 1,
+    'block_number': constants.GENESIS_BLOCK_NUMBER,
+    'gas_limit': constants.GENESIS_GAS_LIMIT,
+    'extra_data': constants.GENESIS_EXTRA_DATA,
+    'nonce': constants.GENESIS_NONCE
+}
+
+
+def genesis_state(setup: Iterable[AddressSetup]) -> Any:
+
+    state = {}
```

Comment:
Not a big deal but we have a nice decorator for doing this in a non-state mutating way.

```python
from eth_utils import to_dict

@to_dict
def make_a_dict():
    yield 'key-a', 1
    yield 'key-b', 2

>>> make_a_dict()
{'key-a': 1, 'key-b': 2}
```

Reply:
I like that :+1: